src/core/parser.zsh

```diff
@@ -168,3 +168,37 @@ _fzf_complete_parse_option_arguments() {
 
     echo - $result
 }
+
+_fzf_complete_parse_option() {
+    local result=()
+    local short=$1
+    local long=$2
+    shift 2
+
+    local shorts=${(pj::)${${(z)short}#*-}}
+    local longs=${(pj:|:)${(z)long}}
+
+    local cmd=(${(Q)${(z)@}})
+    local idx=1
+    while [[ $idx -le ${#cmd} ]]; do
+        if [[ -n $shorts ]]; then
+            if [[ ${cmd[idx]} =~ ^-([^\-$shorts]*[$shorts]+[^$shorts]*)+$ ]]; then
```

Comment:
How about a case where a short option before `$shorts` requires an argument? Consider the following:
```zsh
_fzf_complete_parse_option '-p' '--paginate' 'git -Ccitoppig'
```
In this case, the option `-C` requires an argument so `citoppig` is its argument; however, `_fzf_complete_parse_option` does not know whether `-C` requires an argument and results in `-C`, `-c`, `-i`, `-t`, `-o`, `-p`, `-p`, `-i`, and `-g` being parsed out.

Reply:
BTW, this regex is not hard to understand. Don't worry about its complexity!